src/components/Toolbelt/Toolbelt.js

```diff
@@ -74,7 +78,12 @@ export default function Consumer(props) {
   return (
     <FarmhandContext.Consumer>
       {({ gameState, handlers }) => (
-        <Toolbelt {...{ ...gameState, ...handlers, ...props }} />
+        <Toolbelt
+          fieldMode={gameState.fieldMode}
+          shovelUnlocked={gameState.shovelUnlocked}
+          handleFieldModeSelect={handlers.handleFieldModeSelect}
```

Comment:
seeing the entire gameState and handlers destructured was a bit confusing for me, so i would much rather be explicit here. again though, open to whatever style the codebase prefers

Reply:
In this case I feel pretty strongly about reverting this back to the previous style, it's a pattern that can be easily copy/pasted across the component files and generally not have to be though about too much. I'm actually surprised that you had to make any changes to the `default` export at all here. Was that a stylistic choice, or was it necessary to get things to work?

Reply:
stylistic. your reasoning is sorta exactly why i prefer this, "not have to be thought about" is a bit of a trigger for me. you specifically also know the game code very intimately, but i did not know what props were coming from which object, and this multiple destructuring syntax you're using feels like it's trying to avoid passing things by reference accidentally, but it makes me very confused as to what props are coming into a component. i would be surprised if there aren't also performance implications from doing this vs passing only what is needed. so, my main reasons for wanting to change this are that it's not explicit the way it is now, and it's non-standard for the react community. 

there's actually a much bigger discussion here though because you use the destructure syntax a lot, and i have never seen it used so often. i would really prefer all JSX pass in specific props, it's supposed to look and work like HTML more than javascript from what i understand. you can choose to nip this discussion in the bud though of course as the ultimate code owner here.

one last thing i do want to point out though is what i called the "multiple destructuring".

this is the code that currently exists:
```<Toolbelt {...{ ...gameState, ...handlers, ...props }} />```

but this is what i would expect if the component actually wanted all the items in each of these objects:
```<Toolbelt { ...gameState} {...handlers} {...props } />```

the extra hash you create feels like a performance issue for sure, and that gameState object is huge, i just can't get over that. 😅

Reply:
There's definitely a bigger discussion to be had here! This is a part of the project's architecture that I definitely went my own way with, and I can see how that would create challenges for others. For what it's worth, I never actually thought anyone else would ever be working on this with me, and if I did I might have made some different choices. 😅

In any case, This pattern effectively gives us a sort of "connected component" in the Redux sense, but with pure React APIs. However, there are drawbacks. There is a performance penalty with this spread-ed object approach, but it hasn't manifested into anything significant yet. The bigger issue is the nonstandard nature of it, as you pointed out. I am open to rethinking this and even updating the codebase wholesale to a different, more standard pattern. That said, I'd like to avoid introducing it for one special case in a feature PR like this. If you'd like to change this aspect of the project's architecture, please open a ticket and we'll figure something out. I'm pretty open-minded with making the code more accessible to others, I just want to keep individual PRs as focused as possible. 🙂

>you use the destructure syntax a lot, and i have never seen it used so often. i would really prefer all JSX pass in specific props, it's supposed to look and work like HTML more than javascript from what i understand.

I have the opposite preference, but it is just a preference. I'm fine with you using the HTML-ish syntax where you'd like if that makes things easier for you!

>this is the code that currently exists:
>`<Toolbelt {...{ ...gameState, ...handlers, ...props }} />`
>
>but this is what i would expect if the component actually wanted all the items in each of these objects:
>`<Toolbelt { ...gameState} {...handlers} {...props } />`

This is a really interesting point. It's certainly possible that I got this wrong, and this would be an easy fix if so. We should definitely experiment to see if this can be improved! That said, I'd like to save that for another PR as well so that we can keep changes isolated and easily revert-able if ever necessary.

Reply:
> I never actually thought anyone else would ever be working on this with me, and if I did I might have made some different choices.

ha! this made me laugh, because i know exactly what you mean here

>  I am open to rethinking this and even updating the codebase wholesale to a different, more standard pattern. That said, I'd like to avoid introducing it for one special case in a feature PR like this.

this makes a lot of sense to me. i appreciate the discussion here and i'm going to revert these stylistic changes. we can have a discussion about the tradeoffs as things grow. i'm glad you have the performance concern in mind as well most importantly, and the code is consistent now which is often more valuable than anything to be honest.